src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,822 @@ def test_rotate_img_metadata(self):
+
+        # force an IndexError due to wrong DA shape
+        with self.assertRaises(IndexError):
+            img.getBoundingBox(new_da)
+
+
+class TestAssembleZCube(unittest.TestCase):
+
+    def setUp(self):
+        # create a 2D size and metadata with necessary items
+        self.size = (3200, 3200)
+        self.md = {
+            model.MD_DIMS: 'YX',
+            model.MD_PIXEL_SIZE: (2e-06, 4e-05),
+            model.MD_POS: (0.2, 0.2)
+        }
+
+    def test_assemble_zcube_normal(self):
+        """
+        Verify that using a 2D shaped image DA's in a list will return the right ZCube assembly.
+        """
+        img_list = []
+        z_list = [0.1, 0.2, 0.3, 0.4, 0.5, 0.5, 0.4, 0.3, 0.2, 0.1]
+
+        # create a list of DA
+        for x in range(len(z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, self.size), self.md))
+
+        output_DA = img.assembleZCube(img_list, z_list)
+
+        # check if the input DA shapes are different compared to the output DA
+        self.assertNotEqual(img_list[0].ndim, output_DA.ndim)
+        # check if the Z dimension has the right size
+        self.assertEqual(output_DA.shape[0], len(z_list))
+
+    def test_assemble_zcube_3d(self):
+        """
+        Verify that using a 3D shaped image DA's in a list will fail the
+        assembly of a ZCube and see if a ValueError exception is raised.
+        """
+        # adjust the size for 3D DA
+        self.size = (10, 700, 700)
+
+        img_list = []
+        z_list = [0.1, 0.2, 0.3, 0.4, 0.5, 0.5, 0.4, 0.3, 0.2, 0.1]
+
+        # create a list of DA
+        for x in range(len(z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, self.size), self.md))
+
+        with self.assertRaises(ValueError):
+            img.assembleZCube(img_list, z_list)
+
+    def test_assemble_zcube_z_order(self):
+        """
+        Verify that using a 3D shaped image DA's in a list will return a
+        different ZCube assembly if the order of the images is changed.
+        See if the differences with a list generated before and after reordering are picked up.
+        """
+        img_list = []
+        z_list = [0.1, 0.2, 0.3, 0.4, 0.5]
+
+        # create a list of DA
+        for x in range(len(z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, self.size), self.md))
+
+        output_da_before = img.assembleZCube(img_list, z_list)
+
+        # create a temporary reversed list based on img_list
+        rev_list = img_list.copy()
+
+        # reverse the order of the img_list
+        for index, num in enumerate(reversed(range(len(img_list)))):
+            rev_list[index] = img_list[num]
+
+        output_da_after = img.assembleZCube(rev_list, z_list)
+
+        # check if the Z dimension has the right size
+        self.assertEqual(output_da_before.shape[0], len(z_list) and output_da_after.shape[0], len(z_list))
```

Comment:
I'm pretty sure it's not as you would it to happen. The second argument is `len(z_list) and output_da_after.shape[0]`, which is computed by python as `output_da_after.shape[0]`. The third argument is shown in case of failure.
=> use 2 `self.assertEqual()`